pkgs/build-support/setup-hooks/generate-ld-cache.sh

```diff
@@ -0,0 +1,42 @@
+# shellcheck shell=bash
+
+fixupOutputHooks+=('if [ -z "${dontGenerateLDCache-}" ]; then generateLDCache "$prefix"; fi')
+
+generateLDCache() {
```

Comment:
why not just use patchelf to list the dependencies rather than this bash ?
(does patchelf list the needed transitive? if not, you can write a tiny C binary using https://github.com/lief-project/LIEF)

Might be more maintainable.